minidump/src/minidump.rs

```diff
@@ -1703,6 +1730,15 @@ impl<'mdmp, Descriptor> MinidumpMemoryListBase<'mdmp, Descriptor> {
             .map(|&index| &self.regions[index])
     }
 
+    pub fn read_memory_at_address<T>(&self, addr: teb::Addr64<T>) -> Option<T>
```

Comment:
Giving this method the same name as the one on `MinidumpMemoryBase` but with a slightly different argument feels odd to me. I like the typed pointer offset, though! We could use the same pattern for `RVA` / `MINIDUMP_LOCATION_DESCRIPTOR` if we wanted to be more thorough.